src/main/java/uk/gov/hmcts/reform/idam/web/model/AbstractRegisterRequest.java

```diff
@@ -0,0 +1,34 @@
+package uk.gov.hmcts.reform.idam.web.model;
+
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import uk.gov.hmcts.reform.idam.web.validation.Email;
+
+import javax.validation.constraints.NotEmpty;
+import javax.validation.constraints.Pattern;
+
+@Data
+@EqualsAndHashCode
+abstract class AbstractRegisterRequest {
```

Comment:
Not sure about this - arguably there are more common fields that would belong in this class if you're refactoring, and it won't be clear in the future why they're not included. Having the pattern annotations duplicated in the POJOs wouldn't be a big deal, so you could either move the constants to something shared, or maybe move the regex into a spring property.
Or - if we do want to refactor then we could look into just having one registration class that is consistent about whether to use "username" or "email", but that seems like a lot of effort

Reply:
I'll revert it.